src/connectors/balancer/balancer.ts

```diff
@@ -0,0 +1,331 @@
+      poolId?: string,
+    ) {
+        logger.info(
+            `Fetching pair data for ${quoteToken.address}-${baseToken.address}.`
+          );
+
+          
+        const filter = {
+          where: {
+            id: {
+              eq: poolId,
+            },
+          }
+        };
+    
+        await this.balancer.swaps.fetchPools(poolId? filter: undefined);
+  
+        const info = await this.balancer.swaps.findRouteGivenOut(
+          {
+              tokenIn: quoteToken.address,
+              tokenOut: baseToken.address,
+              amount: amount,
+              gasPrice: BigNumber.from(this._chain.gasPrice.toFixed(0)),
+          }
+        );
+        if (info.swaps.length === 0) {
+          throw new UniswapishPriceError(
+            `No pool found for ${quoteToken.address} to ${baseToken.address}.`
+          );
+        }
+        const marketSp = math.fraction(info.marketSp) as math.Fraction;
+        const executionPrice = new Fraction(marketSp.d.toString(), marketSp.n.toString())
+        return {
+          trade: {
+              swap: {
+                  swapInfo: info,
+                  maxSlippage: this.getAllowedSlippage(allowedSlippage),
+                  deadline: '0',  // updated before trade execution
+                  kind: SwapType.SwapExactOut,
+              },
+              executionPrice
+          },
+          expectedAmount: CurrencyAmount.fromRawAmount(<Currency>quoteToken, info.returnAmount.toString()),
+        };
+    }
+  
+    /**
+     * Given the amount of `baseToken` to put into a transaction, calculate the
+     * amount of `quoteToken` that can be expected from the transaction.
+     *
+     * This is typically used for calculating token sell prices.
+     *
+     * @param baseToken Token input for the transaction
+     * @param quoteToken Output from the transaction
+     * @param amount Amount of `baseToken` to put into the transaction
+     * @param allowedSlippage (Optional) Fraction in string representing the allowed slippage for this transaction
+     */
+    async estimateSellTrade(
+      baseToken: Token,
+      quoteToken: Token,
+      amount: BigNumber,
+      allowedSlippage?: string,
+      poolId?: string
+    ) {
+      logger.info(
+        `Fetching pair data for ${quoteToken.address}-${baseToken.address}.`
+      );
+      
+      const filter = {
+        where: {
+          id: {
+            eq: poolId,
+          },
+        }
+      };
+
+      await this.balancer.swaps.fetchPools(poolId? filter: undefined);
+
+      const info = await this.balancer.swaps.findRouteGivenIn(
+        {
+            tokenIn: baseToken.address,
+            tokenOut: quoteToken.address,
+            amount: amount,
+            gasPrice: BigNumber.from(this._chain.gasPrice.toFixed(0)),
+        }
+      );
+      if (info.swaps.length === 0) {
+        throw new UniswapishPriceError(
+          `No pool found for ${quoteToken.address} to ${baseToken.address}.`
```

Comment:
this should be `No pool found for ${baseToken.address} to ${quoteToken.address}.`